crates/extricrate/src/lib.rs

```diff
@@ -44,13 +64,174 @@ pub mod dependencies {
+                vec![UseStatementType::Alias(full_name, rename.to_string())]
+            }
+
+            UseTree::Glob(UseGlob { .. }) => {
+                vec![UseStatementType::WildCard(prefix.to_owned())]
+            }
+
+            UseTree::Group(UseGroup { items, .. }) => items
+                .iter()
+                .flat_map(|subtree| flatten_use_tree(prefix, subtree))
+                .collect(),
+        }
+    }
+
+    #[derive(Debug, Error)]
+    pub enum ListUseStatementError {
+        #[error("file not found")]
+        FileNotFound,
+        #[error("file not parsable")]
+        FileNotParsable,
+        #[error("file not readable")]
+        FileNotReadable,
+        #[error("path is not a crate")]
+        PathIsNotACrate,
+        #[error("linked module does not exists: {0}")]
+        ModuleDoesNotExists(String),
+    }
+
+    pub fn get_crate_entrypoint(crate_root: &Path) -> Option<PathBuf> {
+        // TODO: support custom entry point
+
+        let cargo_toml = crate_root.join("Cargo.toml");
+        if !cargo_toml.exists() {
+            return None;
+        }
+
+        let main_rs = crate_root.join(Path::new("src/main.rs"));
+        if main_rs.exists() {
+            return Some(main_rs);
+        }
+
+        let lib_rs = crate_root.join(Path::new("src/lib.rs"));
+        if lib_rs.exists() {
+            return Some(lib_rs);
+        };
+        None
+    }
+
     /// List all the `use` statements in the crate, by file/module.
-    pub fn list_use_statements(crate_root: &std::path::Path) -> UseStatementMap {
-        todo!()
+    pub fn list_use_statements(
+        crate_root: &Path,
+    ) -> Result<UseStatementMap, ListUseStatementError> {
+        let mut files_visited = HashSet::new();
+        let mut files_to_visit = VecDeque::new();
+        let entry_point =
+            get_crate_entrypoint(crate_root).ok_or(ListUseStatementError::PathIsNotACrate)?;
+        let src_folder = entry_point
+            .parent()
+            .expect("Failed to get entry point parent folder");
+        files_to_visit.push_back(entry_point.clone());
+        while let Some(file_to_visit) = files_to_visit.pop_front() {
+            if files_visited.contains(&file_to_visit) {
+                continue;
+            }
+
+            if !file_to_visit.exists() {
+                return Err(ListUseStatementError::FileNotFound);
+            }
+
+            let content = read_to_string(&file_to_visit)
+                .map_err(|_| ListUseStatementError::FileNotReadable)?;
+
+            let parsed_file =
+                parse_file(&content).map_err(|_| ListUseStatementError::FileNotParsable)?;
+
+            let mut visitor = UseVisitor::new();
+            visitor.visit_file(&parsed_file);
+            for dependency in visitor.dependencies {
+                match dependency {
+                    UseStatementType::Simple(name) => {
+                        if is_local_import(&name) {
```

Comment:
Oh, interesting, you're following the imports. I fear that may miss some files (a module that's part of the public interface of a library but that's not used in the current crate). We can either walk all the files (easiest) or walk the module declarations (more correct).

But we can do that in a second PR, this one is big enough, I think.